src/dataMigration.js

```diff
@@ -126,4 +126,43 @@ const dataMigrations = [
       });
     },
   },
+  {
+    version: '2.1.0-rc7',
+    migrate: (database) => {
+      const TEMP_STATUS = 'nw';
+      const matchBatch = (batches, batchToMatch) =>
+        batches.find(batch => batch.id === batchToMatch.id);
+      // It was possible to finalise stocktake twice, causing the creating of duplicate
+      // TransactionBatches in inventoryAdjustement, need to clean them up to avoid ledger
+      // discrepancies, const lines = all inventoryAdjustemet transactionItems
+      const lines = database.objects('TransactionItem')
+                            .filtered('transaction.otherParty.type == "inventory_adjustment"');
+      lines.forEach(line => {
+        let batchesToDelete = [];
+        const { batches } = line;
+        // Add duplicated batch to batchesToDelete
+        batches.forEach(batch => {
+          if (!matchBatch(batchesToDelete, batch)) {
+            batchesToDelete = [
+              ...batchesToDelete,
+              ...batches.filtered('itemBatch.id == $0 AND id != $1', batch.itemBatch.id, batch.id),
+            ];
+          }
+        });
+        // Delete batches
+        if (batchesToDelete.length > 0) deleteBatches(line.transaction, batchesToDelete);
+      });
+      // If we try to delete a transactionBatch whos parent transaction is 'confirmed'
```

Comment:
The stocktake is finalised but the inventoryAdjustments are 'confirmed' as per `createRecord -> createInventoryAdjustment`.

I had to check -> In mSupply Desktop it looks like finalised stocktakes have associated inventoryAdjustments as confirmed, looks like we are keeping with consistency

Reply:
Correct, inventory adjustments coming from a finalised stocktake are automatically confirmed in mSupply (means they actually adjust the stock - a suggested status inventory adjustment (if it could exist - it can't) would not affect stock.

Reply:
@GaryWilletts do they never become finalised? Seems odd that they wouldn't but if that's how it be then this PR is fine.

Reply:
They can be finalised manually but are created as confirmed from a finalised stocktake. The reasoning for that is back in the mists of time but maybe so that comments could be added to them before they get fixed for ever.